src/services/obs/backend.rs

```diff
@@ -0,0 +1,310 @@
+
+        let context = HashMap::from([
+            ("bucket".to_string(), bucket.to_string()),
+            ("endpoint".to_string(), endpoint.to_string()),
+        ]);
+
+        let client = HttpClient::new();
+
+        let mut signer_builder = Signer::builder();
+        if let (Some(access_key), Some(secret_key)) = (&self.access_key, &self.secret_key) {
+            signer_builder.access_key(access_key).secret_key(secret_key);
+        }
+
+        // Bucket name must be set in CanonicalizedResource.
+        signer_builder.bucket(&auth_bucket);
+
+        let signer = signer_builder
+            .build()
+            .map_err(|e| other(BackendError::new(context, e)))?;
+
+        info!("backend build finished: {:?}", &self);
+        Ok(Backend {
+            client,
+            root,
+            endpoint,
+            signer: Arc::new(signer),
+            bucket,
+        })
+    }
+}
+
+// Backend for Huaweicloud OBS services.
+#[derive(Debug, Clone)]
+pub struct Backend {
+    client: HttpClient,
+    root: String,
+    endpoint: String,
+    signer: Arc<Signer>,
+    bucket: String,
+}
+
+impl Backend {
+    pub(crate) fn get_abs_path(&self, path: &str) -> String {
+        util::get_abs_path(&self.root, path)
+    }
+}
+
+#[async_trait]
+impl Accessor for Backend {
+    fn metadata(&self) -> AccessorMetadata {
+        let mut am = AccessorMetadata::default();
+        am.set_scheme(Scheme::OBS)
+            .set_root(&self.root)
+            .set_name(&self.bucket)
+            .set_capabilities(None);
+
+        am
+    }
+
+    async fn create(&self, args: &OpCreate) -> Result<()> {
+        todo!()
+    }
+
+    async fn read(&self, args: &OpRead) -> Result<BytesReader> {
+        let p = self.get_abs_path(args.path());
+
+        let resp = self.get_object(&p, args.offset(), args.size()).await?;
+        match resp.status() {
+            StatusCode::OK | StatusCode::PARTIAL_CONTENT => Ok(Box::new(resp.into_body())),
+            _ => {
+                let er = parse_error_response(resp).await?;
+                let err = util::parse_error("read", args.path(), er);
+                Err(err)
+            }
+        }
+    }
+
+    async fn write(&self, args: &OpWrite) -> Result<BytesWriter> {
+        todo!()
+    }
+
+    async fn stat(&self, args: &OpStat) -> Result<ObjectMetadata> {
+        todo!()
+    }
+
+    async fn delete(&self, args: &OpDelete) -> Result<()> {
+        todo!()
+    }
+
+    async fn list(&self, args: &OpList) -> Result<DirStreamer> {
+        todo!()
+    }
+}
+
+impl Backend {
+    pub(crate) async fn get_object(
+        &self,
+        path: &str,
+        offset: Option<u64>,
+        size: Option<u64>,
+    ) -> Result<isahc::Response<isahc::AsyncBody>> {
+        let url = if self.bucket.is_empty() {
+            format!("{}/{}", self.endpoint, percent_encode_path(path))
+        } else {
+            format!(
+                "{}.{}/{}",
```

Comment:
Endpoint should be `https://example.com`, so this format is incorrect.

Reply:
We should construct the correct endpoint during build.

For example:

https://github.com/datafuselabs/opendal/blob/da22720c50dd27cd7375438abc248bce6aad5835/src/services/s3/backend.rs#L743-L748

Reply:
fixed